Fix Extension Docs

Nokel81 opened the pull request:
- fix spelling
- add protocol handler to tree view
- make all lines a single sentance

Signed-off-by: Sebastian Malton <sebastian@malton.name>

Will merge this for the 4.2.0 release but don't want to block the rc.2 release on this.

Nokel81 committed:
fix spelling, add protocol handler to tree view, make all lines a single sentance

Signed-off-by: Sebastian Malton <sebastian@malton.name>

jim-docker commented on docs/extensions/testing-and-publishing/testing.md:
```diff
@@ -41,19 +44,23 @@ test("click called navigate()", () => {
 
-If you are using the [Yeoman Lens Extension Generator](https://github.com/lensapp/generator-lens-ext) to scaffold your extension project. The testing environment [Jest](https://jestjs.io/) are setup for you. Just use  `npm start` or `yarn test` to run the tests.
+If you are using the [Yeoman Lens Extension Generator](https://github.com/lensapp/generator-lens-ext) to scaffold your extension project.
+The testing environment [Jest](https://jestjs.io/) are setup for you.
```
```suggestion
If you are using the [Yeoman Lens Extension Generator](https://github.com/lensapp/generator-lens-ext) to scaffold your extension project then the [Jest](https://jestjs.io/) testing environment is set up for you.
```

jim-docker commented on docs/extensions/testing-and-publishing/testing.md:
```diff
@@ -41,19 +44,23 @@ test("click called navigate()", () => {
 
 ## Main Process Unit Testing
 
 Code in the extension main process are just normal JavaScript files that has access to extension api, you can write unit tests using any testing framework.
```
```suggestion
Code in the extension's main process consists of normal JavaScript files that have access to extension api, you can write unit tests using any testing framework.
```

jim-docker commented on docs/extensions/testing-and-publishing/testing.md:
```diff
@@ -2,9 +2,12 @@
 ## Renderer Process Unit Testing
 
-UI components in extension renderer process are based on React/ReactDOM. These components can be tested by popular React testing tools like [React Testing Library](https://github.com/testing-library/react-testing-library).
+UI components in extension renderer process are based on React/ReactDOM.
```
```suggestion
UI components in the extension's renderer process are based on React/ReactDOM.
```

jim-docker commented on docs/extensions/testing-and-publishing/testing.md:
```diff
@@ -2,9 +2,12 @@
 
-If you are using the [Yeoman Lens Extension Generator](https://github.com/lensapp/generator-lens-ext) to scaffold extension project. The testing environment for render process are already setup for you. Just use `npm start` or `yarn test` to run the tests.
+If you are using the [Yeoman Lens Extension Generator](https://github.com/lensapp/generator-lens-ext) to scaffold extension project.
+The testing environment for render process are already setup for you.
```
```suggestion
If you are using the [Yeoman Lens Extension Generator](https://github.com/lensapp/generator-lens-ext) to scaffold your extension project then the testing environment for the renderer process is already set up for you.
```

Nokel81 committed:
fix a few wording issues

Signed-off-by: Sebastian Malton <sebastian@malton.name>

Update docs/extensions/testing-and-publishing/testing.md

Co-authored-by: Jim Ehrismann <40840436+jim-docker@users.noreply.github.com>

Update docs/extensions/testing-and-publishing/testing.md

Co-authored-by: Jim Ehrismann <40840436+jim-docker@users.noreply.github.com>

jim-docker commented on docs/extensions/testing-and-publishing/testing.md:
```diff
@@ -2,11 +2,10 @@
 
-If you are using the [Yeoman Lens Extension Generator](https://github.com/lensapp/generator-lens-ext) to scaffold extension project.
-The testing environment for render process are already setup for you.
+If you are using the [Yeoman Lens Extension Generator](https://github.com/lensapp/generator-lens-ext) to scaffold extension project then the testing environment for render process is already setup for you.
```
```suggestion
If you are using the [Yeoman Lens Extension Generator](https://github.com/lensapp/generator-lens-ext) to scaffold extension project then the testing environment for render process is already set up for you.
```

jim-docker commented on docs/extensions/testing-and-publishing/testing.md:
```diff
@@ -49,10 +48,9 @@ Extend your tests based on the examples.
 
-If you are using the [Yeoman Lens Extension Generator](https://github.com/lensapp/generator-lens-ext) to scaffold your extension project.
-The testing environment [Jest](https://jestjs.io/) are setup for you.
+If you are using the [Yeoman Lens Extension Generator](https://github.com/lensapp/generator-lens-ext) to scaffold your extension project then the [Jest](https://jestjs.io/) is setup for you.
```
```suggestion
If you are using the [Yeoman Lens Extension Generator](https://github.com/lensapp/generator-lens-ext) to scaffold your extension project then the [Jest](https://jestjs.io/) testing environment is set up for you.
```

jim-docker reviewed (commented):
You didn't take my suggestions verbatim :)
